context.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	//"errors"
+
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/urfave/cli"
+	yaml "gopkg.in/yaml.v2"
+)
+
+var contextsPath string
+var defaultContextPath string
+
+func contextCmd() cli.Command {
+	return cli.Command{
+		Name:  "context",
+		Usage: "manage context",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Aliases:   []string{"c"},
+				Usage:     "create a new context",
+				ArgsUsage: "<context> <provider>",
+				Action:    create,
+				Flags: []cli.Flag{
+					cli.StringSliceFlag{
+						Name:  "config",
+						Usage: "context configuration",
+					},
+				},
+			},
+			{
+				Name:    "list",
+				Aliases: []string{"l"},
+				Usage:   "list contexts",
+				Action:  list,
+			},
+			{
+				Name:      "set",
+				Aliases:   []string{"s"},
+				Usage:     "set context for future invocations",
+				ArgsUsage: "<context>",
+				Action:    set,
+			},
+		},
+	}
+}
+
+// EnsureConfiguration ensures context configuration directory hierarchy is in place, if not
+// creates it and the default context configuration files
+func EnsureConfiguration() {
```

Comment:
Maybe this should be in the config.go file.

Reply:
Agreed, also makes sense to potentially not include this context.go file as it provides no value for this PR if EnsureConfiguration() is moved?